israel_bank_account_validator.py

```diff
@@ -1,168 +1,309 @@
 import logging
 from typing import Union
 
+# Setting up logging
+logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
-
 SUPPORTED_BANKS = {
     'YAHAV': 4,
-    'POST': 9,
+    'ISRAEL_POST': 9,
     'LEUMI': 10,
     'DISCOUNT': 11,
     'HAPOALIM': 12,
     'IGUD': 13,
-    'OTSAR_AHAYAL': 14,
+    'OTSAR_HAHAYAL': 14,
     'MERCANTILE': 17,
+    'ONE_ZERO': 18,
     'MIZRAHI_TEFAHOT': 20,
-    'CITIBANK': 22,
+    'CITYBANK': 22,
+    'HSBC': 23,
     'BEINLEUMI': 31,
     'ARAVEI_ISRAELI': 34,
     'MASAD': 46,
-    'POALEI_AGUDAT_ISRAEL': 52
+    'POALEI_AGUDAT_ISRAEL': 52,
+    'JERUSALEM': 54
 }
 
+# Constants
+MIZRAHI_TEFAHOT_BRANCH_THRESHOLD = 400
+MIZRAHI_TEFAHOT_BRANCH_THRESHOLD_MIN = 401
+MIZRAHI_TEFAHOT_BRANCH_THRESHOLD_MAX = 799
+
+LEUUMI_BRANCH_THRESHOLD = 800
+
+
+def convert_to_int(input_value):
+    if isinstance(input_value, str):
+        return int(input_value)
```

Comment:
Please add try catch

Reply:
@fabian-cellosign We have exception handling on the caller side and it's in purpose: https://github.com/republixapps/israel-bank-account-validator/pull/1/files#diff-82c6de86245fefe2ebdd0cfaa669bc7e97e658324f9f264516ff71115e9ca636R82